src/storage/src/source/reclock/compat.rs

```diff
@@ -112,49 +113,60 @@ where
         // Allow manually simulating the scenario where the since of the remap
         // shard has advanced too far.
         fail_point!("invalid_remap_as_of");
-        assert!(
-            PartialOrder::less_equal(since, &as_of),
-            "invalid as_of: as_of({as_of:?}) < since({since:?}), \
-            source {id}, \
-            remap_shard: {:?}",
-            metadata.remap_shard
-        );
-
         assert!(
             as_of.elements() == [IntoTime::minimum()] || PartialOrder::less_than(&as_of, upper),
             "invalid as_of: upper({upper:?}) <= as_of({as_of:?})",
         );
 
-        tracing::info!(
-            ?since,
-            ?as_of,
-            ?upper,
-            "{operator}({id}) {worker_id}/{worker_count} initializing PersistHandle"
-        );
-
-        use futures::stream;
-        let events = stream::once(async move {
-            let updates = read_handle
-                .snapshot_and_fetch(as_of.clone())
-                .await
-                .expect("since <= as_of asserted");
-            let snapshot = stream::once(std::future::ready(ListenEvent::Updates(updates)));
-
-            let listener = read_handle
-                .listen(as_of.clone())
-                .await
-                .expect("since <= as_of asserted");
-
-            let listen_stream = stream::unfold(listener, |mut listener| async move {
-                let events = stream::iter(listener.fetch_next().await);
-                Some((events, listener))
+        let events = if since.is_empty() {
```

Comment:
Did you find this problem because of the more eager since downgrades that we're now doing in the controller, upon collection creation?